aepp/destinationinstanceservice.py

```diff
@@ -0,0 +1,81 @@
+import logging
+from .configs import ConnectObject
+
+class Instance:
+    """
+    This class is referring to Destination Instance Service capability for AEP.
+    """
+    def __init__(self,
+        config: Union[dict,ConnectObject] = aepp.config.config_object,
+        header: dict = aepp.config.header,
+        loggingObject: dict = None,
+        **kwargs,):
+        """
+        Instanciating the class for destination instance service
+
+        Arguments:
+            loggingObject : OPTIONAL : logging object to log messages.
+            config : OPTIONAL : config object in the config module.
+            header : OPTIONAL : header object  in the config module.
+        possible kwargs:
+        """
+        if loggingObject is not None and sorted(
+            ["level", "stream", "format", "filename", "file"]
+        ) == sorted(list(loggingObject.keys())):
+            self.loggingEnabled = True
+            self.logger = logging.getLogger(f"{__name__}")
+            self.logger.setLevel(loggingObject["level"])
+            if type(loggingObject["format"]) == str:
+                formatter = logging.Formatter(loggingObject["format"])
+            elif type(loggingObject["format"]) == logging.Formatter:
+                formatter = loggingObject["format"]
+            if loggingObject["file"]:
+                fileHandler = logging.FileHandler(loggingObject["filename"])
+                fileHandler.setFormatter(formatter)
+                self.logger.addHandler(fileHandler)
+            if loggingObject["stream"]:
+                streamHandler = logging.StreamHandler()
+                streamHandler.setFormatter(formatter)
+                self.logger.addHandler(streamHandler)
+        if type(config) == dict: ## Supporting either default setup or passing a ConnectObject
+            config = config
+        elif type(config) == ConnectObject:
+            header = config.getConfigHeader()
+            config = config.getConfigObject()
+        self.connector = connector.AdobeRequest(
+            config_object=config,
+            header=header,
+            loggingEnabled=self.loggingEnabled,
+            logger=self.logger,
+        )
+        self.header = self.connector.header
+        # self.header.update({"Accept": "application/json"})
+        self.header.update(**kwargs)
+        if kwargs.get('sandbox',None) is not None: ## supporting sandbox setup on class instanciation
+            self.sandbox = kwargs.get('sandbox')
+            self.connector.config["sandbox"] = kwargs.get('sandbox')
+            self.header.update({"x-sandbox-name":kwargs.get('sandbox')})
+            self.connector.header.update({"x-sandbox-name":kwargs.get('sandbox')})
+        else:
+            self.sandbox = self.connector.config["sandbox"]
+        self.endpoint = config.endpoints["global"] + config.endpoints["destinationInstance"]
+        
+        
+    def createAdHocDatasetExport(self, adhocObj: dict = None)->dict:
```

Comment:
It would be great to have just individual parameters instead of the whole free-form dict so we can just call the function without having to lookup the expected syntax for that dict - for example it would be more convenient to use `createAdhocDatasetExport(dataset_id="foo", flow_id="bar")`

Reply:
Hi @cmenguy, this endpoint can take multiple flows and multiple datasets at a time. for example, export dataset1 to flow1, export dataset2 and dataset3 to flow2 in one request. We can provide this simple version of one flow and one dataset and expand it as needed.

Reply:
Ok makes sense, could we just have the dict then be flow to datasets and have the function take care of converting it to whatever format is needed by the API? For example if we could pass `{"flow1": ["dataset1"], "flow2": ["dataset2", "dataset3"]}` this will help avoid the need to know the syntax of the API.

Reply:
Added suggested change and unit test for the transformation. @cmenguy